locator/tablets.cc

```diff
@@ -370,6 +370,10 @@ future<> tablet_metadata::set_colocated_table(table_id id, table_id base_id) {
         _table_groups.erase(it);
     }
 
+    if (!_table_groups.contains(base_id)) {
+        on_internal_error(tablet_logger, format("Trying to set co-located table {} with base table {} but it's not a base table.", id, base_id));
```

Comment:
A not-so-side question about CDC. As far as I know, user writes __are allowed__ for CDC log tables. I checked and I was able to write a row, I also checked that you can use LWT to do that - and it is, indeed, not forbidden. I am not entirely sure of the reason why writes were allowed, but I think the intention was so that you can partially delete the contents of the CDC log table. @elcallio or @nyh, do you perhaps remember? Perhaps we should also forbid LWT for CDC log tables.